src/directives/onLongPress.ts

```diff
@@ -0,0 +1,137 @@
+import {ElementPart, nothing} from 'lit';
+import {AsyncDirective, directive, PartType} from 'lit/async-directive.js';
+import type {
+  DirectiveParameters,
+  DirectiveClass,
+  DirectiveResult,
+  PartInfo
+} from 'lit/async-directive.js';
+
+type LongPressCallback = (event?: PointerEvent) => void;
+
+const DEFAULT_LONG_PRESS_TIMEOUT_MS = 1000;
+
+class LongPressDirective extends AsyncDirective {
+  /** Element of the directive. */
+  #element!: Element;
```

Comment:
the linter should be disallowing non-null assertions really. can you make this `#element?: Element`

since it won't actually always be defined